### ISSUE TITLE
noam learning rate scheme buggy since 1.4.3

### ISSUE THREAD

**Issue by martinpopel** [tensorflow/tensor2tensor#555]
Before 1.4.3, the "noam" learning rate scheme used to have a linear warmup followed by an exponential decay.
In https://github.com/tensorflow/tensor2tensor/commit/9b52264e08989d050d5270ea2770622541bd76cb#diff-0a47b7977216c77d4a17cf56db6dfc11L298 this default was changed, most probably unintentionally.
Now, instead of calling `optimize.learning_rate_decay` directly, the `optimize.learning_rate_warmup(warmup_steps, warmup_schedule="exp")` method is called, which in turn calls `optimize.learning_rate_decay`.
In general, it is a good idea to refactor the source code, so warmup can be configured independently on the rest of the learning rate scheme (now called "decay", although it could be e.g. cyclic or constant, thus non-decaying function).
However, the current implementation has several bugs:

* The default warmup scheme was changed from `linear` to `exp` without notifying the users and without creating `transformer_base_v3` (and keeping `transformer_base_v2` hparams intact for replication of old experiments). The change may be beneficial for some experiments (although according to my preliminary experiments it is not), but changing the defaults in each version makes working with T2T frustrating (my current advice for my colleagues is *don't update T2T unless you really have to, in which case don't expect to replicate your old experiments*).
* There is no way how to override the warmup scheme from `exp` to `linear` using hparams because `learning_rate_warmup` is [called](https://github.com/tensorflow/tensor2tensor/blob/103d0571ef48d82b61aae67cd61d716738f9fcfe/tensor2tensor/utils/optimize.py#L219) without the optional parameter.
* The "noam" scheme implementation was kept [the same](https://github.com/tensorflow/tensor2tensor/blob/103d0571ef48d82b61aae67cd61d716738f9fcfe/tensor2tensor/utils/optimize.py#L171-L173) including the warmup, which is misleading because now the main warmup code is elsewhere.
* There interplay of the two implementations of the warmup (exp and linear) creates unexpected bumps:
![learning-rate-bug](https://user-images.githubusercontent.com/724617/35782332-8ce6a3ae-09f6-11e8-978f-32a7ab1479f1.png)
The orange curve is the default "noam" with learning_rate=0.2 and warmup=16k and it looks as expected. The blue curve is "noam" with learning_rate=0.2 and warmup=32k.

**Comment by vince62s** [comment]
Changing "exp" to "linear" is obviously not enough.


![image](https://user-images.githubusercontent.com/15141326/35809250-5fa66004-0a88-11e8-994d-81a00ed367c2.png)

**Comment by rsepassi** [comment]
Will be fixed in 1.4.4, coming out today. Sorry about that.

**Comment by rsepassi** [comment]
Fixed. Please let us know if you encounter issues again.
